dask/array/linalg.py

```diff
@@ -650,9 +666,14 @@ def compression_matrix(data, q, n_power_iter=0, seed=None, compute=False):
     q: int
         Size of the desired subspace (the actual size will be bigger,
         because of oversampling, see ``da.linalg.compression_level``)
+    iterator: {'none', 'power', 'QR'}, default='none'
+        Define the technique used for iterations to cope with flat
+        singular spectra or when the input matrix is very large.
     n_power_iter: int
         number of power iterations, useful when the singular values of
         the input matrix decay very slowly.
+    n_oversamples: int, default=10
+        Number of oversamples used for generating the sampling matrix.
```

Comment:
The Halko paper mentions that "As a practical matter, setting p = 5 or p = 10 often gives superb results".  What do you think of adding a comment here to that effect?  Perhaps:

```suggestion
        Number of oversamples used for generating the sampling matrix.
        This value increases the size of the subspace computed, which is more 
        accurate at the cost of efficiency.  Results are rarely sensitive to this choice
        though and in practice a value of 10 is very commonly high enough.
```

I think it's important that most users know that this is not a parameter they would likely want to change.

Reply:
Well, my problem with the setting as it was, was that I wanted to control it myself and be able to reduce it (that's why I propagated it). But indeed, the p=5  or p=10 is a nice suggestion!